Adjust error messages

pakrym-oai opened the pull request:
<img width="1378" height="285" alt="image" src="https://github.com/user-attachments/assets/f0283378-f839-4a1f-8331-909694a04b1f" />

pakrym-oai committed:
Adjust error messages

pakrym-oai committed:
core/client.rs: simplify flatten usage with and_then for plan_type extraction
login/lib.rs: remove unnecessary clippy::unwrap_used expect attribute

pakrym-oai committed:
error: update usage limit messages in tests for clarity and reset timing details

bolinfest commented on codex-rs/core/src/error.rs:
```diff
@@ -115,6 +117,30 @@ pub enum CodexErr {
+        {
+            write!(
+                f,
+                "You've hit your usage limit. Upgrade to Pro (https://openai.com/chatgpt/pricing), or wait for limits to reset (every 5h and every week)"
```
Should we put a period at the end of this message and also the one in the `else` case? Feels a little weird for there to be one sentence with one and one without?

bolinfest commented on codex-rs/login/src/lib.rs:
```diff
@@ -132,15 +131,23 @@ impl CodexAuth {
+            .and_then(|t| t.id_token.chatgpt_plan_type.clone())
+    }
+
+    fn get_current_auth_json(&self) -> Option<AuthDotJson> {
+        #[expect(clippy::unwrap_used)]
+        self.auth_dot_json.lock().unwrap().clone()
+    }
```
I was in the middle of trying to make `AuthDotJson` private to the login crate. Do we need to open this up?

pakrym-oai replied:
this is not public method, right?

bolinfest reviewed (commented):
We are on a bit of a collision course for `codex-rs/login/src/lib.rs` with respect to my current stack.

pakrym-oai committed:
Merge branch 'main' into pakrym/adjust-error-messages

pakrym-oai committed:
error.rs: improve punctuation and consistency in error messages
login: return PlanType as lowercase string in get_plan_type
token_data: add PlanType::as_string to get string representation
